### PR TITLE
feat!: rewrite with nuxt kit and nuxt 3 support

### PR DESCRIPTION
**Pull request by maciejpedzich**

### CODE REVIEW

File: .editorconfig
Comment (pi0) [inline .editorconfig]: Why this file is removed?

File: .eslintignore
Comment (pi0) [inline .eslintignore:1]: Same for this why is removed?

File: .github/workflows/ci.yml
Comment (pi0) [inline .github/workflows/ci.yml]: Why is this file removed?

File: CHANGELOG.md
Comment (pi0) [inline CHANGELOG.md]: Please keep changelog history! It is continuation of module!

File: LICENSE
Comment (pi0) [inline LICENSE]: Copyright shall remain for Nuxt Community rather than individual name. Also I guess you forgot to resolve merge conflict

File: README.md
Comment (pi0) [inline README.md]: I'm not sure this issue is affecting end-users

File: README.md
Comment (pi0) [inline README.md]: It can be simply `@nuxtjs/feed`

File: README.md
Comment (pi0) [inline README.md]: We use `modules` as default in Nuxt 3

File: build.config.ts
Comment (pi0) [inline build.config.ts:1]: This shouldn't be needed when using module-builder

File: docs/docus.config.ts
Comment (pi0) [inline docs/docus.config.ts]: TODO: To update before merge

File: docs/package.json
Comment (pi0) [inline docs/package.json]: Please avoid individual name.

File: docs/package.json
Comment (pi0) [inline docs/package.json:12]: Should be devDependency.

File: docs/pages/2.setup.md
Comment (pi0) [inline docs/pages/2.setup.md]: We use `modules` in nuxt 3

File: docs/pages/4.differences.md
Comment (pi0) [inline docs/pages/4.differences.md:9]: /cc To check with @manniL

File: package.json
Comment (pi0) [inline package.json]: We should use `@nuxtjs/feed` and `v3.

File: playground/feed/array.ts
Comment (pi0) [inline playground/feed/array.ts]: Let's use fetch and say goodby to axios :)

File: renovate.json
Comment (pi0) [inline renovate.json:1]: Why removed?

File: src/module.ts
Comment (pi0) [inline src/module.ts]: `build:before` hook not needed. We can instead invoke `feed:extend` to allow extending feed sources and avoid awaitable config in nuxt.config.

File: src/module.ts
Comment (pi0) [inline src/module.ts]: With using nuxt/kit, `nuxt` is already bound and can be accessed with `useNuxt()` from `@nuxt/kit`

File: src/module.ts
Comment (pi0) [inline src/module.ts]: Would be nice to find a way to fetch sources without blocking Nuxt lifecycle

**Review by pi0 — CHANGES_REQUESTED** [review]

**Comment by pi0** [comment]
Thanks for the pull request @maciejpedzich. I've made an overall review this is in a nice direction.

- It is fine to generally have customizations in repo setup. But ideally, it should be as close as possible with [module starter](https://github.com/nuxt/starter/tree/module) to keep it's consistency with the rest of (new) nuxt-community modules and I would like to hear from @manniL preferences as he is the current maintainer.
- Since we are rewriting the module with Nuxt 3 in mind, ideally should also follow its design goals. Mainly, integration with nitro with an API route to allow runtime generation of sitemap vs on build generation. We can do it later but have it in mind. In the meantime, Avoid async resolve function in nuxt.config and depend on nuxt hooks for extension. (added in review comments)

File: docs/docus.config.ts
Comment (TheAlexLichter) [inline docs/docus.config.ts]: Same as below :)

File: test/module.test.js
Comment (TheAlexLichter) [inline test/module.test.js:3]: We definitely need some kind of test modules as well.

**Comment by TheAlexLichter** [comment]
Thanks for the PR! I'll have a look in the next days ☺️

**Comment by maciejpedzich** [comment]
Hi @pi0, thanks for the review! Sorry for taking so long, luckily I've found time to properly get down to adjusting the code accordingly with the changes you've requested. I'll let you when I'm done correcting everything.

File: .editorconfig
Comment (maciejpedzich) [reply .editorconfig]: Restored!

File: .eslintignore
Comment (maciejpedzich) [reply .eslintignore:1]: Restored this one too

File: .github/workflows/ci.yml
Comment (maciejpedzich) [reply .github/workflows/ci.yml]: Not anymore!

File: CHANGELOG.md
Comment (maciejpedzich) [reply CHANGELOG.md]: Brought it back and added a provisional 3.0.0 entry

File: LICENSE
Comment (maciejpedzich) [reply LICENSE]: Sorry, force of habit! That conflict should be resolved now

File: README.md
Comment (maciejpedzich) [reply README.md]: This one's long been resolved, but the warning stayed there for whatever reason. It's gone

File: README.md
Comment (maciejpedzich) [reply README.md]: You've got it, chief!

File: README.md
Comment (maciejpedzich) [reply README.md]: Copy!

File: build.config.ts
Comment (maciejpedzich) [reply build.config.ts:1]: TypeScript says otherwise, it's not that huge of an overhead anyway

File: docs/package.json
Comment (maciejpedzich) [reply docs/package.json]: Force of habit as I said. Replaced!

File: docs/docus.config.ts
Comment (maciejpedzich) [reply docs/docus.config.ts]: I don't understand this one

File: docs/package.json
Comment (maciejpedzich) [reply docs/package.json:12]: I don't understand this one either, it's totally valid in my opinion

File: docs/pages/2.setup.md
Comment (maciejpedzich) [reply docs/pages/2.setup.md]: Roger that!

File: package.json
Comment (maciejpedzich) [reply package.json]: Changed it as well

File: playground/feed/array.ts
Comment (maciejpedzich) [reply playground/feed/array.ts]: Your wish is my command - all references to axios should now be replaced with ohmyfetch

File: renovate.json
Comment (maciejpedzich) [reply renovate.json:1]: Restored

File: src/module.ts
Comment (maciejpedzich) [reply src/module.ts]: Copy, made use of it in generateFeedFile

File: src/module.ts
Comment (maciejpedzich) [reply src/module.ts]: I can't get `feed:extend` to work properly. Either I've missed something in the docs, or they don't mention anything like that anywhere.

File: src/module.ts
Comment (maciejpedzich) [reply src/module.ts]: No sure on that one, I'm afraid

File: docs/docus.config.ts
Comment (maciejpedzich) [reply docs/docus.config.ts]: Replaced it

File: test/module.test.js
Comment (maciejpedzich) [reply test/module.test.js:3]: Will take care of these later

File: .github/workflows/ci.yml
Comment (TheAlexLichter) [inline .github/workflows/ci.yml:36]: Is there a specific reason for using `npm` here? ☺️

File: .github/workflows/ci.yml
Comment (TheAlexLichter) [inline .github/workflows/ci.yml:41]: It'd be great to release a tested module (even if not *fully* tested).

File: docs/pages/1.index.md
Comment (TheAlexLichter) [inline docs/pages/1.index.md:15]:
```suggestion
- API similar to <a href="https://github.com/nuxt-community/feed-module"><code>@nuxt-community/feed-module</code></a>, making migration over to <code>@nuxtjs/feed</code> painless
```

File: docs/pages/2.setup.md
Comment (TheAlexLichter) [inline docs/pages/2.setup.md:15]:
```suggestion
npm install @nuxtjs/feed
```

File: docs/pages/3.usage.md
Comment (TheAlexLichter) [inline docs/pages/3.usage.md:41]: I'd suggest printing the feed object here:

```
feed: {
  async create() { /** the rest **/ }
}
```

File: docs/pages/3.usage.md
Comment (TheAlexLichter) [inline docs/pages/3.usage.md:42]: Could we use ohmyfetch or (node) fetch for the example?

File: docs/pages/3.usage.md
Comment (TheAlexLichter) [inline docs/pages/3.usage.md:61]: 'WET' might not be clear to everyone. Maybe use something like "repeatable"?

File: src/lib/resolveModuleOptions.ts
Comment (TheAlexLichter) [inline src/lib/resolveModuleOptions.ts:8]: No else needed. Same below.

File: src/module.ts
Comment (TheAlexLichter) [inline src/module.ts:27]: Now that SSG landed, the `feed` module could provide static feed generation directly too 😋

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]
Left a few comments! Good job so far ☺️

**Comment by maciejpedzich** [comment]
@manniL, @pi0 - I've decided to retire my custom Nuxt 3 compatible RSS feed module, and I'm afraid I won't be able to actively support the feed module as a community maintainer because I simply don't have the bandwidth to do it anymore. Feel free to reuse the code as a foundation for the official one.

**Comment by TheAlexLichter** [comment]
> @manniL, @pi0 - I've decided to retire my custom Nuxt 3 compatible RSS feed module, and I'm afraid I won't be able to actively support the feed module as a community maintainer because I simply don't have the bandwidth to do it anymore. Feel free to reuse the code as a foundation for the official one.

@maciejpedzich Thanks a lot for being that open and for the PR! I will try to move the module to Nuxt3 based on your code in near future (and give appropriate credits) 👍

**Comment by entity** [comment]
> > @manniL, @pi0 - I've decided to retire my custom Nuxt 3 compatible RSS feed module, and I'm afraid I won't be able to actively support the feed module as a community maintainer because I simply don't have the bandwidth to do it anymore. Feel free to reuse the code as a foundation for the official one.
> 
> @maciejpedzich Thanks a lot for being that open and for the PR! I will try to move the module to Nuxt3 based on your code in near future (and give appropriate credits) 👍

Hey folks, is there any working sitemap generator these days that supports Nuxt 3 and Nuxt Content V2?

**Comment by xlanex6** [comment]
Hi @manniL 

Is there a roadmap, todo list to finish the job initiate by @maciejpedzich ? 
I will need RSS feed on my Nuxt 3 project so...  I'm ready to spend time and energy to finish what have to be done.

**Comment by TheAlexLichter** [comment]
Feel free to check #106. I haven't had the time to continue from here, so whoever wants to take a stab is good to go ☺️

**Pull request closed without merging**
